src/layouts/Header.tsx

```diff
@@ -10,52 +10,59 @@ import styles from './Header.less'
 
 const { Header } = Layout
 
-class HeaderView extends PureComponent {
-  state = {
-    visible: true
-  }
+function HeaderView (props) {
+  const [visible, setVisible] = useState(true)
+  // TO Verify: set logic for oldScrollTop and ticking
+  const [oldScrollTop, setOldScrollTop] = useState(document.body.scrollTop)
+  const [ticking, setTicking] = useState(true)
 
-  static getDerivedStateFromProps (props, state) {
-    if (!props.autoHideHeader && !state.visible) {
-      return {
-        visible: true
-      }
+  useEffect(() => {
+    // componentdidmount
+    document.addEventListener('scroll', handScroll, { passive: true })
+    // Specify how to clean up after this effect:
+    return function cleanup () {
+      // componentdidunmount
+      document.removeEventListener('scroll', handScroll)
     }
-    return null
-  }
+  })
 
-  componentDidMount () {
-    document.addEventListener('scroll', this.handScroll, { passive: true })
-  }
-
-  componentWillUnmount () {
-    document.removeEventListener('scroll', this.handScroll)
-  }
+  const { isMobile, handleMenuCollapse, setting } = props
+  const { navTheme, layout, fixedHeader } = setting
 
-  getHeadWidth = () => {
-    const { isMobile, collapsed, setting } = this.props
+  const getHeadWidth = () => {
+    const { isMobile, collapsed, setting } = props
     const { fixedHeader, layout } = setting
     if (isMobile || !fixedHeader || layout === 'topmenu') {
       return '100%'
     }
     return collapsed ? 'calc(100% - 80px)' : 'calc(100% - 256px)'
   }
 
-  handleNoticeClear = type => {
+  // TODO: cant see usage
+  // const getDerivedStateFromProps = (props) => {
+  //   if (!props.autoHideHeader && !visible) {
+  //     return {
+  //       visible: true
+  //     }
+  //   }
+  //   return null
+  // }
```

Comment:
the `getDerivedStateFromProps` is a special function - part of the `Component` API, [details](getDerivedStateFromProps).
So what you should do is to declare a local variable:
```
let visible
```
And update accordingly.

- [ ] correctly initialize `visible` variable.